src/toast/ops/conviqt.py

```diff
@@ -658,3 +658,97 @@ def _exec(self, data, detectors=None, **kwargs):
                 timer.report_clear(f"conviqt process detector {det}")
 
         return
+
+
+
+class SimTEBConviqt(SimConviqt):
+    """
+    Operator that uses libconviqt to generate beam-convolved timestreams.
+    This operator should be used in presence of a spinning HWP which makes the beam time-dependent,
+    constantly mapping the co- and cross-polar responses on to each other.
+    In the parent class OpSimConviqt we assume the beam to be static.
+
+
+    The convolution  is performed by  coupling each IQU component of the signal propertly as:
+    :math:`skyT_lm * beamT_lm, skyE_lm * Re{P}, skyB_lm * Im{P}`.
+
+    For extra details please refer to [this note ](https://giuspugl.github.io/reports/Notes_TEB_convolution.html)
+
+"""
+
+    @function_timer
+    def _exec(self, data, detectors=None, **kwargs):
+        if not self.available:
+            raise RuntimeError("libconviqt is not available")
+
+        if self.comm is None:
+            raise RuntimeError("libconviqt requires MPI")
+
+        if self.detector_pointing is None:
+            raise RuntimeError("detector_pointing cannot be None.")
+
+        log = Logger.get()
+
+        timer = Timer()
+        timer.start()
+
+        # Expand detector pointing
+        self.detector_pointing.apply(data, detectors=detectors)
+
+        all_detectors = self._get_all_detectors(data, detectors)
+
+        for det in all_detectors:
+            verbose = self.comm.rank == 0 and self.verbosity > 0
+
+            # Expand detector pointing
+            self.detector_pointing.apply(data, detectors=[det])
+
+            if det in self.sky_file_dict:
+                sky_file = self.sky_file_dict[det]
+            else:
+                sky_file = self.sky_file.format(detector=det, mc=self.mc)
+            sky = self.get_sky(sky_file, det, verbose)
+
+            beam_file = self.beam_file.format(detector=det, mc=self.mc)
+            beam_T = self.get_beam(beam_file.replace(".fits","_T.fits"), det, verbose)
+            beam_E = self.get_beam(beam_file.replace(".fits","_E.fits"), det, verbose)
+            beam_B = self.get_beam(beam_file.replace(".fits","_B.fits"), det, verbose)
+
+            detector = self.get_detector(det)
+
+            theta, phi, psi, psi_pol = self.get_pointing(data, det, verbose)
+
+            # I-beam convolution
+            pnt = self.get_buffer(theta, phi, psi, det, verbose)
+
+            convolved_data = self.convolve(sky, beam_T, detector, pnt, det, verbose)
+
+            del pnt,
+            # Q-beam convolution
```

Comment:
I presume the comment should say `E-beam convolution`?

Reply:
It is the convolution with the E-beam, resulting in the Q signal

Reply:
Sorry for interrupting with stupid questions: but don't you need both E and B components to compute the Q signal? There is no one-to-one correspondence between E and Q (and B and U), only between E/B and Q/U together, correct?
If you have a pointer where I can read more about this, I'd be really grateful.

Reply:
Correct the sky E and B alms are convolved with the _polarized _ beam  and both participate to the Q and U  signals. 
have a look at  [this note](https://giuspugl.github.io/reports/Notes_TEB_convolution.html) for some reference

Reply:
Thanks!
So if I understand correctly this still makes some assumptions about the beam, and if we ever wnt to tackle the fully general problem, we have to look at the expressions in https://arxiv.org/abs/2012.10437v1 . I'm still working on that, but I honestly have no idea when that will be ready...
How far will the current approximations take us in terms of LiteBIRD simulations? Will they be good enough for all practical purposes?

Reply:
The improvements apply mainly to polarization data , now we can include the polarized component of the grasp beam and do not assume it to be the same as the unpolarized one.  As a result this approach yields to lower side lobe pick up than the latter approximation .